python/all-your-base/all_your_base.py

```diff
@@ -0,0 +1,25 @@
+def rebase(input_base, digits, output_base):
+    if output_base <= 1:
+        raise ValueError("output base must be greater than 1")
+    if input_base <= 1:
+        raise ValueError("input base must be greater than 1")
+    if len(list(filter(lambda x: x != 0, digits))) == 0:
+        return []
+
+    # Build up the value in base 10 first to convert into new base
+    base_10_val = 0
+    for i in range(len(digits)):
+        if digits[i] < 0:
+            raise ValueError("digits must all be positive or zero")
+        if digits[i] >= input_base:
+            raise ValueError("digits must be less than the input base")
+        base_10_val += digits[i] * (input_base ** (len(digits) - i - 1))

```

Comment:
If you use the built-in `enumerate` function you can rewrite this whole loop as a generator comprehension

```python
n = len(digits)
base_10 = sum(d * (input ** (n - i - 1)) for i, d in enumerate(digits))
```
If you choose not to write it as a comprehension using `enumerate` will still help with making the loop more Pythonic. In my experience `range(len(digits))` is usually not what you want.